spec/install/deploy_spec.rb

```diff
@@ -295,6 +295,106 @@
       expect(out).not_to include("You have deleted from the Gemfile")
     end
 
+    it "allows to replace a host with the same host with credentials when allow_deployment_source_credential_changes is true" do
+      gemfile <<-G
+        source "https://user_name:password@some.host.com/gems/path"
+        gem "rack"
+      G
+
+      lockfile <<-G
+        GEM
+          remote: https://some.host.com/gems/path/
+          specs:
+            rack (1.0.0)
+
+        PLATFORMS
+          #{local}
+
+        DEPENDENCIES
+          rack
+      G
+
+      bundle! "config allow_deployment_source_credential_changes true"
+      bundle "install --deployment"
```

Comment:
this needs to use `forgotten_command_line_option(:deployment => true)` rather than passing the flag directly -- see the surrounding specs for an example